plugins/outputs/timestream/timestream.go

```diff
@@ -275,11 +343,39 @@ func init() {
 
 func (t *Timestream) Write(metrics []telegraf.Metric) error {
 	writeRecordsInputs := t.TransformMetrics(metrics)
+
+	errs := make(chan error, len(writeRecordsInputs))
+	var wg sync.WaitGroup
+	wg.Add(len(writeRecordsInputs))
+
+	start := time.Now()
 	for _, writeRecordsInput := range writeRecordsInputs {
-		if err := t.writeToTimestream(writeRecordsInput, true); err != nil {
-			return err
-		}
+		go func(inp *timestreamwrite.WriteRecordsInput) {
+			defer wg.Done()
+			if err := t.writeToTimestream(inp, true); err != nil {
+				errs <- err
+			}
+
+		}(writeRecordsInput)
+	}
+
+	wg.Wait()
+	now := time.Now()
+	elapsed := now.Sub(start)
+
+	close(errs)
+
+	t.Log.Infof("##WriteToTimestream - Metrics size: %d request size: %d time(ms): %d",
+		len(metrics), len(writeRecordsInputs), elapsed.Milliseconds())
+
+	// On partial failures, Telegraf will reject the entire batch of metrics and
+	// retry. writeToTimestream will return retryable exceptions only.
+	err, _ := <-errs
```

Comment:
On line 355, we are only adding to errs channel if err != nil
Reading it n times currently just blocks forever when error does not happen.

I can try removing the != nil check and see if it works un-interupted

Reply:
Yes, that sounds reasonable, to remove the nil check and always return the result from writeToTimestream.

Reply:
I have added a range over channel. It exists on the first encountered error. But since the channel is already closed before that, it should be able to garbage collected?